code/modules/paperwork/paperbin.dm

```diff
@@ -21,83 +28,145 @@
 		if(temp && !temp.is_usable())
-			to_chat(user, "<span class='notice'>You try to move your [temp.name], but cannot!</span>")
-			return
-	var/response = ""
-	if(!papers.len > 0)
-		response = alert(user, "Do you take regular paper, or Carbon copy paper?", "Paper type request", "Regular", "Carbon-Copy", "Cancel")
-		if (response != "Regular" && response != "Carbon-Copy")
-			add_fingerprint(user)
+			to_chat(user, SPAN_NOTICE("You try to move your [temp.name], but cannot!"))
 			return
-	if(amount >= 1)
-		amount--
-		if(amount==0)
-			update_icon()
-
-		var/obj/item/paper/P
-		if(papers.len > 0)	//If there's any custom paper on the stack, use that instead of creating a new paper.
-			P = papers[papers.len]
-			papers.Remove(P)
-		else
-			if(response == "Regular")
+
+	if(LAZYLEN(papers) < 1 && amount < 1)
+		to_chat(user, SPAN_WARNING("[src] is empty!"))
+		return
+
+	var/obj/item/paper/P
+	if(LAZYLEN(papers) > 0)	//If there's any custom paper on the stack, use that instead of creating a new paper.
+		P = papers[papers.len]
+		LAZYREMOVE(papers, P)
+	else
+		var/paper_kind = input(user, "What kind of paper?") in list("Regular", "Green", "Blue", "Pink", "Yellow", "Carbon-Copy", "Cancel")
+		switch(paper_kind)
+			if("Regular")
 				P = new /obj/item/paper
-				if(global.current_holiday?.name == "April Fool's Day")
-					if(prob(30))
-						P.info = "<font face=\"[P.crayonfont]\" color=\"red\"><b>HONK HONK HONK HONK HONK HONK HONK<br>HOOOOOOOOOOOOOOOOOOOOOONK<br>APRIL FOOLS</b></font>"
-						P.rigged = 1
-						P.updateinfolinks()
-			else if (response == "Carbon-Copy")
+			if("Green")
+				P = new /obj/item/paper/green
+			if("Blue")
+				P = new /obj/item/paper/blue
+			if("Pink")
+				P = new /obj/item/paper/pink
+			if("Yellow")
+				P = new /obj/item/paper/yellow
+			if("Carbon-Copy")
 				P = new /obj/item/paper/carbon
-		user.put_in_hands(P)
-		to_chat(user, "<span class='notice'>You take [P] out of the [src].</span>")
-	else
-		to_chat(user, "<span class='notice'>[src] is empty!</span>")
+			else
+				return
 
+		if(!istype(P, /obj/item/paper/carbon) && global.current_holiday?.name == "April Fool's Day" && prob(30))
+			P.rigged = TRUE
+			P.set_content("<font face=\"[P.crayonfont]\" color=\"red\"><b>HONK HONK HONK HONK HONK HONK HONK<br>HOOOOOOOOOOOOOOOOOOOOOONK<br>APRIL FOOLS</b></font>")
+
+	user.put_in_hands(P)
+	to_chat(user, SPAN_NOTICE("You take \the [P] out of \the [src]."))
+	amount--
+	update_icon()
 	add_fingerprint(user)
 	return
 
+/obj/item/paper_bin/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/paper))
+		if(amount >= max_amount)
+			to_chat(user, SPAN_WARNING("\The [src] is full!"))
+			return
+		if(!user.unEquip(I, src))
+			return
+		add_paper(I)
+		to_chat(user, SPAN_NOTICE("You put [I] in [src]."))
+		return TRUE
 
-/obj/item/paper_bin/attackby(obj/item/i, mob/user)
-	if(istype(i, /obj/item/paper))
-		if(!user.unEquip(i, src))
+	else if(istype(I, /obj/item/paper_bundle))
+		if(amount >= max_amount)
```

Comment:
you only check if there's at least 1 empty slot here, but put all papers in separately, this can overflow the bin

Reply:
Yes? So it will only add what fits in?
That's intended, so refilling isn't a pain in the ass.